watchtower/standalone.go

```diff
@@ -142,6 +152,39 @@ func (w *Standalone) Stop() error {
 	return nil
 }
 
+// createNewHiddenService automatically sets up a v2 or v3 onion service in
+// order to listen for inbound connections over Tor.
+func (w *Standalone) createNewHiddenService() error {
+	// Get all the ports the watchtower is listening on. These will be used to
+	// map the hidden service's virtual port.
+	listenPorts := make([]int, 0, len(w.listeners))
+	for _, listener := range w.listeners {
+		port := listener.Addr().(*net.TCPAddr).Port
+		listenPorts = append(listenPorts, port)
+	}
+
+	// Once we've created the port mapping, we can automatically create the
+	// hidden service. The service's private key will be saved on disk in order
+	// to persistently have access to this hidden service across restarts.
+	onionCfg := tor.AddOnionConfig{
+		VirtualPort: DefaultPeerPort,
+		TargetPorts: listenPorts,
+		Store:       tor.NewOnionFile(w.cfg.WatchtowerKeyPath, 0600),
+		Type:        w.cfg.Type,
+	}
+
+	addr, err := w.cfg.TorController.AddOnion(onionCfg)
```

Comment:
Would be nice to log the onion address created.

Reply:
👍

Reply:
Perhaps the `tor` package should log this in `AddOnion` instead? And similar things upon authentication success / failure?

Reply:
I think this should be deferred to another PR that adds logging to the `tor` package.

Reply:
Fine with me, should we still add the log here though?

Reply:
yup, probably also in the other place in `server.go` too ?

Reply:
Ah I didn't realize we didn't log it there either. Not a big deal then.

Reply:
Will make an issue about this so it's at least documented